### DISCUSSION TITLE
Failing Windows

### DISCUSSION THREAD

**Discussion by FireResearch-BK** [firemodels/fds#12049] [Q&A] [answered]
We have a model with lots of windows. Is there a more elegant way to fail them at a certain temperature due to thermal damage than placing a HOLE, triggered via control logic by a temperature sensor, at each window?

We've considered enabling BURN_AWAY for the SURF, but does not look like an option as the window must not contribute any cobustible mass to the fire. We simply need it to fail to open up a passage to the ambient domain.

**Comment by drjfloyd** [comment, marked as answer]
For a reaction on &MATL, there is the limitation that the sum of NU_MATL, NU_SPEC, and NU_PART must be less than or equal to 1. That is you can't make more mass than present in the solid. If it sums to less than 1 than any unaccounted for mass just disappears. You could define a MATL reaction with a reference temperautre and a narrow peak width that has no NU_XXXX values with BURN_AWAY, but with this approach the window would disappear cell-by-cell and not all at once. If you want the entire widow to be removed at once, then you would need a HOLE or OBST with a DEVC_ID or CTRL_ID for each window.

**Reply by FireResearch-BK** [reply]
Well, I think a gradual opening by cell could be explained by a gradual melting of the window.

Is there any experience with the CPU time penalty for MATL reactions compared to, say multiple devices and holes?

**Comment by drjfloyd** [comment]
Negligible additional cost for the MATL reaction for the windows

**Comment by FireResearch-BK** [comment]
Thanks!

Looks like I'll give a MATL reaction a try.

**Comment by FireResearch-BK** [comment]
I ran a few tests and the MATL reaction seems to work. So thanks for the help, @drjfloyd!

Here is a simple test case:
[Window_Burn_Test.txt](https://github.com/firemodels/fds/files/12357916/Window_Burn_Test.txt)

What I'm still trying to figure out is which parameter - other than lowering material density - can be adjusted to have the glass start burning away fairly early (between 20 or 40 seconds in the simulation).

**Comment by mcgratta** [comment]
`REFERENCE_TEMPERATURE`

**Comment by IFAB-AS** [comment]
We did many tests with BURN_AWAY of glass in the last weeks. Yesterday, I tried to reproduce the test results from this paper https://publications.iafss.org/publications/fss/6/1063/view/fss_6-1063.pdf. It is basically a heated wall that generates a heat flux between 0-10 kW/m² to a glass window.
We used various REFERENCE_TEMPERATURE and PYROLYSIS_RANGE settings. Some results of the test are shown in the image below.

![grafik](https://github.com/firemodels/fds/assets/124035126/10154702-c9f2-4484-a8d6-bf1534bbb42c)

As the "cracking" cannot be directly compared to the "hole" in FDS, we assume that the resulting curve should be similar to the cracking curve, but might be shifted to larger times.

The following results can be seen from this:
- Increasing T_ref (REFERENCE_TEMPERATURE) moves the curves to the right or up in the diagram (PYROLYSIS_RANGE=standard=80)
- Decreasing the PYROLYSIS_RANGE with constant T_Ref moves the curves down or to the left in the diagram
- There exists a range, where no cracking or breaking of the glass occurs (marked in red). It is not possible to find any combination of T_ref and pyroysis_range that represents this behavior. This means: If we increase the PYROLYSIS_RANGE to higher values, the window will react at room temperature and will burn_away after some time. If we increase the T_Ref, the curve does not represent the measurements for higher heat fluxes anymore.
- From previous tests with intumescend materials, we know that the transition can be reproduced by taking a THRESHOLD_TEMPERATURE. This feature is not used in the latest FDS code anymore. From other discussions, I have found that the PYROLYSIS_RANGE should be used instead. But we have seen that it is not possible to create a similar curve with this (most likely also for intumescent materials or other scenarios).

It is possible to simulate the glass breaking also with DEVC temperature measurements for each panel (e.g. temperature difference in glass or absolute temperature criteria), but if we have a large building and a free burn scenario, it is too complex to do this step.

Do you have any other suggestions? Why was the threshold temperature removed?
[Window_Burn_Test.txt](https://github.com/firemodels/fds/files/14387527/Window_Burn_Test.txt)

**Comment by mcgratta** [comment]
If you make `PYROLYSIS_RANGE` very small, you are essentially doing what `THRESHOLD_TEMPERATURE` did. That is, you are saying that the reaction occurs at the reference temperature, but not at lower temps.

**Comment by drjfloyd** [comment]
You can't make PYROLYSIS_RANGE too small.  This uses Eq. 9.5 to get a refrence rate which is then used in 9.4 to get A and E.  If you make PYROLYSIS_RANGE too small for the REFERENCE_TEMPERATURE, then you will overflow A.  Check the .out file to look at the A and E that result and make sure you don't see Inifinty or #N/A there for the material.

**Comment by mcgratta** [comment]
Good point. If you set it to a few degrees C, that should be enough to get a sharp reaction curve. If you make the range large, you will essentially start the reaction at low, even ambient, temperature.

**Comment by johodges** [comment]
What key physics are you trying to capture with predicting the window crack? Are you trying to get small amounts of leakage through the cracks, predict generation of a larger hole, or the window falling out?

**Comment by IFAB-AS** [comment]
We are trying to find an estimation of the failure of the window as this influences the ventilation in the building (e.g. larger hole or failure of the complete window). We also collected the values where the hole is larger than 1 cell in FDS (e.g. 0.5 m x 0.5 x), but these are similar to the shown values.
I will try reducing the Pyrolysis_range next week.

**Comment by johodges** [comment]
I have been investigating a few approaches to modeling failure of polycarbonate windows with a mix of burn through + fallout. For fall out I have been using a device to monitor the back side temperature of the window (either max or mean depending on the case I was considering) then removing the remaining parts of the window once the device activates. An example device:

&DEVC ID='WINDOW_1_T', QUANTITY='WALL TEMPERATURE', SPATIAL_STATISTIC='MEAN', XB=1.96,2.48,1.8,1.8,0.24,0.56, SETPOINT=200., INITIAL_STATE=.TRUE./

**Comment by IFAB-AS** [comment]
I lowered the PYROLYSIS_RANGE to 3 °C (we get Ininifty for lower values) with various temperatures. The green curve with REFERENCE_TEMPERATURE=70 is a good approximation for most of the data points, but the glass will also break for low heat flux values.

![grafik](https://github.com/firemodels/fds/assets/124035126/21297037-f95e-428d-9abf-17d3c1169cf6)


I know from previous projects that the THRESHOLD_TEMPERATURE can be used to fix this behavior for the low range. I can also remember that it was possible to set the THRESHOLD_TEMPERATURE>REFERENCE_TEMPERATURE together with a larger PYROLYSIS_RANGE to get a strong reaction if the threshold is reached. If this cut-off value is used, it might be possible to create a curve that looks similar to the dotted line.

**Comment by mcgratta** [comment]
How about two reactions? Or equivalently, say the glass consists of two components. The first component reacts at a low temperature and creates the second component, which reacts at a higher temperature. This higher temperature should be approximately the adiabatic surface temperature when the heat flux is 5 kw/m2.

**Comment by IFAB-AS** [comment]
From our understanding, you mean something such as:
&SURF ID='Glass',
      RGB=146,152,255,
      TRANSPARENCY=0.501961,
      BURN_AWAY=.TRUE.,
      BACKING='VOID',
      MATL_ID(1,1)='Glass1',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=0.003/

&MATL ID='Glass1',
      SPECIFIC_HEAT=0.8,
      CONDUCTIVITY=1.0,
      DENSITY=2500.0,
      EMISSIVITY=0.1,
	  NU_MATL=1.0,
	  MATL_ID='Glass2',
	  REFERENCE_TEMPERATURE=70.0, PYROLYSIS_RANGE=3/

&MATL ID='Glass2',
      SPECIFIC_HEAT=0.8,
      CONDUCTIVITY=1.0,
      DENSITY=2500,
      EMISSIVITY=0.1,
	  REFERENCE_TEMPERATURE=340.0, PYROLYSIS_RANGE=10/

Glass1 is the standard glass that creates Glass2 at the same place. T_ref is 340°C from the adiabatic surface temperature at 5 kW/m². But why should the 2nd glass burn? Even if we lower the T_ref, it would be basically something like adding the times of two materials for each heat flux (or adding two of curves in the images above).

**Comment by drjfloyd** [comment]
REFERENCE_TEMPERATURE and PYROLYSIS_RANGE use an assumed HEATING_RATE to get A and E.  That heating rate may not be a good assumption at the low heat fluxes. You could try playing with the HEATING_RATE input to see if you can make that curve steeper.

**Comment by IFAB-AS** [comment]
I did some tests with different HEATING_RATEs. Several are still running and mybe there is a good set of parameters in there. If we increase the heating_rate, the materials burn faster. If we decrease the heating rate, we get a steeper curve but also larger times for the higher heat fluxes. 
If we decrease the HEATING_RATE while using a narrow PYROLYSIS_RANGE (T_REF=50 and PYR_RANGE=3, HEATING_RATE=1) and compare this to the same PYROLYSIS_RANGE with the standard HEATING_RATE=5, there is no difference at all. Is there an explanation for this? According to the formulas in the user guide in chapter 9.2.2 the kinetic constant A should still change and the mass loss should be increasing as in our other tests.

**Comment by drjfloyd** [comment]
Try running the cases with  TGA_ANALYSIS=T and setting TGA_HEATING_RATE as needed. If you don't see differences in the virtual TGA tests, then there may be some kind of issue. If you do see differences there then it might just be that timestep sizes and actual heating rate at the larger scale is driving results.

**Comment by IFAB-AS** [comment]
We were able to find values that represent the data points for the glass from the test. The problem was:
- emissivity was used for a low-e glass (modern glass), but the test was made 30 years ago with standard glass
- our "test window" was much larger (what could go wrong?) and the reflection between heater and window was heating up the window
- the DEVC was placed at the other side of the heater to get the "true heater heat flux at the distance of the window". Problem is: The window is contributing to heat flux due to reflections and DEVC is measuring a too low value

The kinetic constants are working correctly, the difference between the heating rates (0.1, 1, 5, 10) is just very small for a low PYROLYSIS_RANGE as the E-parameter dominates the behavior of the mass loss as this is exponentially used in the differential equation.
